src/Microsoft.Framework.CodeGeneration.Templating/Compilation/MetadataReferencesProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.Framework.CodeGeneration.Templating.Compilation
+{
+    public class MetadataReferencesProvider
```

Comment:
Yes. But those references are used solely for compiling the .cs file which the Razor Engine creates from the template. The other references weren't necessary. Have tried this out with all the tests and templates I could find and checked with @PradeepKadubandi and it was fine.